File: src/com/quotegraph/model/SqlConnection.java
package com.quotegraph.model;

import java.sql.Connection;

/**
 * Interface used for database abstraction
 * 
 * @author Daniel
 */
public interface SqlConnection {
    
    /**
     * 
     */
    public void connect();
    
    /**
     * 
     */
    public void close();
    
    /**
     * 
     * @return 
     */
    public Connection getConn();
    
    /**
     * 
     * @return 
     */
    public boolean hasError();
    
}

File: src/main/kotlin/ch/peters/daniel/quotegraph/controller/SqlConnection.kt
package ch.peters.daniel.quotegraph.controller

import ch.peters.daniel.quotegraph.interfaces.ISqlConnection

import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

/**
 * Opens a connection to an SQL server.
 *
 * @author Daniel Peters
 * @version 2.0
 */
class SqlConnection(override var config: DbConfig) : ISqlConnection {
  override lateinit var connection: Connection
  override var error: Boolean = false
  private var link: String = ""

  private fun setLink() {
    link = "jdbc:" + config.type + "://" + config.host + ":" + config.port + "/" + config.database
  }

  override fun connect() {
    try {
      setLink()
      connection = DriverManager.getConnection(link, config.user, config.password)
    } catch (ex: SQLException) {
      error = true
    }
  }

  override fun close() {
    try {
      connection.close()
    } catch (ex: SQLException) {
      System.out.println("An error occurred while closing the connection.")
    }
  }
}
